x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -870,6 +873,15 @@ export class Embeddable
     return this.savedVis && this.savedVis.description;
   }
 
+  public getFilters() {
+    if (!this.savedVis) {
+      throw new Error('savedVis is required for getFilters');
+    }
```

Comment:
Why did we decide to guard this implementation, but default to an empty array in the visualize embeddable?

Reply:
Good point! The `injectFilterReferences` doesn't accept undefined parameters, so that's why I added the `throw` there. But perhaps this is better logic?

```suggestion
    return mapAndFlattenFilters(
      this.deps.injectFilterReferences(
        this.savedVis?.state.filters ?? [],
        this.savedVis?.references ?? []
      )
    );
```

Reply:
I was just pointing out the inconsistency (not necessarily prescribing an approach for your use case), but I think this makes sense! 👍 It looks like the panel gets refreshed when the savedVis becomes available.